test/integration/operator_run_bundle_test.go

```diff
@@ -0,0 +1,235 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from https://github.com/kubernetes-sigs/kubebuilder/tree/39224f0/test/e2e/v3
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-sdk/internal/olm/operator"
+	"github.com/operator-framework/operator-sdk/internal/olm/operator/bundle"
+	testutils "github.com/operator-framework/operator-sdk/test/internal"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	wait "k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+)
+
+// TODO: After run bundle is complete, remove the boolean variable and the if statements.
+var runBundleImplemented = false
+
+var (
+	tc  testutils.TestContext
+	err error
+)
+
+var _ = BeforeSuite(func() {
+	By("creating a new context")
+	// For this test suite, the bundle image name is equal to tc.ImageName
+	tc, err = testutils.NewTestContext("GO111MODULE=on")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(tc.Prepare()).To(Succeed())
+
+	createOperator(tc, defaultOperatorName)
+})
+
+var _ = AfterSuite(func() {
+	By("removing container image and working dir")
+	tc.Destroy()
+})
+
+var _ = Describe("run bundle", func() {
+	It("Run Bundle Basic", func() {
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
+		err := cfg.Load()
+		Expect(err).NotTo(HaveOccurred())
+
+		i := bundle.NewInstall(cfg)
+		i.BundleImage = tc.ImageName
+
+		err = doInstall(i)
+		Expect(err).NotTo(HaveOccurred())
+
+		uninstall(kubeconfigPath, tc)
+	})
+
+	It("Run bundle SingleNamespace", func() {
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath, Namespace: tc.Kubectl.Namespace}
+		err := cfg.Load()
+		Expect(err).NotTo(HaveOccurred())
+
+		i := bundle.NewInstall(cfg)
+		i.BundleImage = tc.ImageName
+		i.InstallMode.InstallModeType = v1alpha1.InstallModeTypeSingleNamespace
+		i.InstallMode.TargetNamespaces = []string{tc.Kubectl.Namespace}
+
+		err = doInstall(i)
+		Expect(err).NotTo(HaveOccurred())
+
+		uninstall(kubeconfigPath, tc)
+	})
+
+	It("Run bundle OwnNamespace", func() {
+		if !runBundleImplemented {
+			Skip("Run bundle not yet implemented")
+		}
+
+		cfg := &operator.Configuration{KubeconfigPath: kubeconfigPath}
```

Comment:
This may not be testing `OwnNamespace` since the default namespace used might not be the one created for this test (`tc.Kubectl.Namespace`).